packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/edge/EdgeLayout.ts

```diff
@@ -40,6 +47,64 @@ export const getUpdatedConnectionHandles: GetUpdatedConnectionHandlesParameters
     }
   );
 
+  const sidedHandlesIds = getHandlesIdsFromNode(sourceNode);
+  const otherEndsCenter: Map<string, XYPosition> = new Map<string, XYPosition>();
+
+  edges.forEach((edge) => {
+    if (edge.sourceNode && edge.targetNode && edge.sourceHandle && edge.targetHandle) {
+      if (sidedHandlesIds.find((id) => id === edge.sourceHandle)) {
+        let node = _nodes.find((n) => n.id === edge.targetNode?.id);
+        if (node) {
+          otherEndsCenter.set(edge.sourceHandle, getNodeCenter(node, _nodes));
+        }
+      }
+      if (sidedHandlesIds.find((id) => id === edge.targetHandle)) {
+        let node = _nodes.find((n) => n.id === edge.sourceNode?.id);
+        if (node) {
+          otherEndsCenter.set(edge.targetHandle, getNodeCenter(node, _nodes));
+        }
+      }
+    }
+  });
+
+  sourceNode.data.connectionHandles
+    .filter((handle) => !handle.virtual)
+    .map((nodeConnectionHandle: ConnectionHandle) => {
+      Object.values(Position).map((position) => {
+        //Update index of handles on the same side
+        if (nodeConnectionHandle.position === position) {
+          let indexCounter = 0;
+          if (nodeConnectionHandle.id) {
+            const currentOtherEndCenter = otherEndsCenter.get(nodeConnectionHandle.id);
+            sourceNode.data.connectionHandles
+              .filter((handle) => handle.position === position && handle.id !== nodeConnectionHandle.id)
+              .filter((handle) => !handle.virtual)
+              .forEach((handle) => {
+                if (currentOtherEndCenter && handle.id && handle.position === position) {
+                  let otherEndCenter = otherEndsCenter.get(handle.id);
+
+                  if (position === Position.Bottom || position === Position.Top) {
+                    if (otherEndCenter && otherEndCenter.x < currentOtherEndCenter.x) {
+                      indexCounter = indexCounter + 1;
+                    } else {
+                      indexCounter = indexCounter - 1;
+                    }
+                  } else {
+                    if (otherEndCenter && otherEndCenter.y < currentOtherEndCenter.y) {
+                      indexCounter = indexCounter + 1;
+                    } else {
+                      indexCounter = indexCounter - 1;
+                    }
+                  }
+                }
+              });
+            nodeConnectionHandle.index = indexCounter;
```

Comment:
Might be correct... but difficult to read/understand, and possibly not as efficient as possible. I'll try and see if I find a better way (it'll also help me understand what this does precisely).